tasks/main.yml

```diff
@@ -72,15 +68,42 @@
 - name: deploy puppet environment with r10k
   command: "/usr/local/bin/r10k deploy environment -c {{puppet_etc_dir}}/r10k.yaml -p -v error {{puppet_environment}}"
 
-- name: generate client certificate for puppetmaster
-  command: "puppet cert --generate {{hostname}}"
-  when: puppet_configured|failed
+- name: generate types for the environments to prevent autoload issues
+  command: "/opt/puppetlabs/bin/puppet generate types --environment {{ puppet_environment }}"
+
+- name: ensure puppet can create the modules directory
+  shell: chown -R puppet:puppet /etc/puppetlabs/code
```

Comment:
ansible suggested using the file module to change directory permission.

Reply:
Good point. Changed that to file module and used the variable to the directory instead of the hard coded one.